typescript/packages/subsurface-viewer/src/layers/wells/utils/spline.ts

```diff
@@ -222,6 +228,64 @@ export function splineRefine(data_in: FeatureCollection): FeatureCollection {
     return data;
 }
 
+/**
+ * Will reduce/coarse the wellpaths.
+ */
+export function coarsenWells(data_in: FeatureCollection): FeatureCollection {
+    const data = cloneDeep(data_in);
+
+    const no_wells = data.features.length;
+
+    for (let well_no = 0; well_no < no_wells; well_no++) {
+        const mds = data.features[well_no].properties?.["md"];
+        if (mds === undefined) {
+            continue;
+        }
+        const geometryCollection = data.features[well_no]
+            .geometry as GeometryCollection;
+        const lineString = geometryCollection?.geometries[1] as LineString;
+
+        if (lineString.coordinates?.length === undefined) {
+            continue;
+        }
+
+        const coords = lineString.coordinates as Position3D[];
+
+        const n = coords.length;
+
+        const newCoordinates: Position3D[] = [];
+        const newMds: number[][] = [];
+        newMds.push([]);
+
+        newCoordinates.push(coords[0]);
+        newMds[0].push(mds[0][0]);
+
+        for (let i = 1; i < n - 1; i += 1) {
+            if (i % 5) {
+                const P1 = coords[i + 0];
+                const md1 = mds[0][i + 0];
+
+                newCoordinates.push(P1);
+                newMds[0].push(md1);
+            }
+        }
```

Comment:
Could you use [lodash's `remove`](https://lodash.com/docs/4.17.15#remove)?

Reply:
Good point. But I need the index of the elements so I used the  standard javascript filter method instead.